planner/core/logical_plan_builder.go

```diff
@@ -2678,6 +2693,68 @@ func (b *PlanBuilder) buildWindowFunction(p LogicalPlan, expr *ast.WindowFuncExp
 	return window, nil
 }
 
+// resolveWindowSpec resolve window specifications for sql like `select ... from t window w1 as (w2), w2 as (partition by a)`.
+// We need to resolve the referenced window to get the definition of current window spec.
+func resolveWindowSpec(spec *ast.WindowSpec, specs map[string]ast.WindowSpec, inStack map[string]bool) error {
+	if inStack[spec.Name.L] {
+		return ErrWindowCircularityInWindowGraph
+	}
+	if spec.Ref.L == "" {
+		return nil
+	}
+	ref, ok := specs[spec.Ref.L]
+	if !ok {
+		return ErrWindowNoSuchWindow.GenWithStackByArgs(spec.Ref.O)
+	}
+	inStack[spec.Name.L] = true
+	err := resolveWindowSpec(&ref, specs, inStack)
+	if err != nil {
+		return err
+	}
+	inStack[spec.Name.L] = false
+	return mergeWindowSpec(spec, &ref)
+}
+
+func mergeWindowSpec(spec, ref *ast.WindowSpec) error {
+	if ref.Frame != nil {
+		return ErrWindowNoInherentFrame.GenWithStackByArgs(ref.Name.O)
+	}
+	if ref.OrderBy != nil {
+		if spec.OrderBy != nil {
+			name := spec.Name.O
+			if name == "" {
+				name = "<unnamed window>"
+			}
+			return ErrWindowNoRedefineOrderBy.GenWithStackByArgs(name, ref.Name.O)
+		}
+		spec.OrderBy = ref.OrderBy
+	}
+	if spec.PartitionBy != nil {
+		return ErrWindowNoChildPartitioning
```

Comment:
Without `GenWithStackByArgs`, the error won't add error stask info. So you may need `errors.Trace()` here if you want the stack.